native-utils/src/main/java/com/github/weisj/darklaf/platform/PointerUtil.java

```diff
@@ -37,6 +39,30 @@
      * @return the handle.
      */
     public static long getHWND(final Component component) {
-        return Pointer.nativeValue(Native.getComponentPointer(component));
+        if (System.getProperty("os.name").toLowerCase().startsWith("mac")) {
+            return getHWNDMacOS(component);
+        } else {
+            return Pointer.nativeValue(Native.getComponentPointer(component));
+        }
     }
+
+    private static long getHWNDMacOS(final Component component) {
+        long handle = Pointer.nativeValue(Native.getComponentPointer(component));
+        if (handle != 0) return handle;
+
+        Window window = SwingUtilities.getWindowAncestor(component);
+        try {
+            Method getPeer = Window.class.getMethod("getPeer");
+            Object peer = getPeer.invoke(window);
+            Method getPlatformWindow = peer.getClass().getMethod("getPlatformWindow");
+            Object platformWindow = getPlatformWindow.invoke(peer);
+            Method getContentView = platformWindow.getClass().getMethod("getContentView");
+            Object contentView = getContentView.invoke(platformWindow);
+            Method getAwtView = contentView.getClass().getMethod("getAWTView");
+            return (Long) getAwtView.invoke(contentView);
```

Comment:
This is equivalent to `sun.lwawt.macosx.CPlatformWindow#getLayerPtr`, and it does not work:

```
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2020-02-23 19:26:12.383 java[50459:22210854] -[CGLLayer styleMask]: unrecognized selector sent to instance 0x7fb6226d0c30
2020-02-23 19:26:12.383 java[50459:22210854] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[CGLLayer styleMask]: unrecognized selector sent to instance 0x7fb6226d0c30'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff40292acd __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff6a994a17 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff4030c8d6 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007fff4023493f ___forwarding___ + 1485
	4   CoreFoundation                      0x00007fff402342e8 _CF_forwarding_prep_0 + 120
	5   libdarklaf-macos.dylib              0x00000001338a4ea5 Java_com_github_weisj_darklaf_platform_macos_JNIDecorationsMacOS_installDecorations + 37
	6   ???                                 0x0000000118e07950 0x0 + 4712331600
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Reply:
So it returns 0?

Reply:
It returns non-null value, however, the resulting value does not seem to be window pointer.
The pointer itself is stored in `CPlatformWindow.ptr` field